Use Hibernate Second-Level Cache for infrastructure entities and users [5]

<!--
Please read the Contributing guidelines (https://github.com/hzi-braunschweig/SORMAS-Project/blob/development/docs/CONTRIBUTING.md) before submitting an issue. You don't have to remove this comment or any other comment from this issue as they will automatically be hidden.
-->
### Problem Description

Currently a lot of referenced entities that rarely change are fetched from the database in most transactions, often several times. This causes always a bit of delay, adding up in complex or bulk functions.

### Proposed Change
<!-- Mandatory -->

We want to improve performance for loading and persisting data. Infrastructure and user entities are broadly used in almost all data access and manipulation actions. Holding them in the cache instead of having to load them again and again should help. 

Using Hibernate L2 Cache (new dependency `hibernate-ehcache`) and defining which JPA entities are allowed to be cached (`@Cacheable`). Tutorial: https://www.baeldung.com/hibernate-second-level-cache

### Acceptance Criteria
<!-- Optional -->

- [x] Use the `PerformanceLogAnalysisGenerator` to create before and after performance logs for saving a case (see https://github.com/hzi-braunschweig/SORMAS-Project/blob/development/docs/TROUBLESHOOTING.md#debugging-performance-problems). 
    - [x] Make sure to save two separate cases that make use of the same users, districts and regions, so we can see the effect of the cache. 
    - [x] Also log all database queries that are executed when saving the case.
    - [x] Also measure performance of `DistrictFacadeEjb.getAllActiveByRegion` without and with cache
- [x] Make sure it has no significant (100MB+) impact on working memory usage. E.g. by adding a helper method in StartupShutdownService that gets all entities of all infrastructure types once.
- [x] Discuss results of performance measurements and memory impact (reviewed by @BarnaBartha and @StefanKock)

### Implementation Details
<!-- Optional -->

- [x] Active Hibernate Second-Level Cache (L2 Cache) and make all entities that inherit InfrastructureAdo and the user entity cache-able with NONSTRICT_READ_WRITE strategy - see https://www.baeldung.com/hibernate-second-level-cache
- [x] Limit maximum cache entries per entity to 5000 for communities, facilities and points of entry
- [x] Adjust `BaseAdoService.getByUuid` to only get the ID of the entity and then use EntityManger.find. This will allow us to make more use of the L2 and L1 cache. 
- [x] Set all ManyToOne relations of the case that use these entities to lazy, so they can make use of the L2 cache. 
- [x] Related to the above point validate the following assumption: With active L2 cache when a Case entity is loaded, a query is created that includes all the fields needed to fill the responsible region entity. When Case.getResponsibleRegion is set to be lazy, those fields are not longer part of the query.
- [x] If setting the ManyToOne relations to infrastructure and user entities of the case works, ~~create an issue to~~ do the same for all other core entities and their children.

### Follow-Up

- [x] Discuss whether we should set `<property name="hibernate.jpa.compliance.proxy" value="false" />` to prevent lazy loaded entities from being directly loaded (see https://stackoverflow.com/questions/61752552/manytoone-failed-to-lazy-load-when-hibernate-second-cache-is-enabled/61769276#61769276) -> discussed by @BarnaBartha and @StefanKock: No change needed.

### Additional Information
<!-- Optional -->
- Using use L2 Cache for entities like `CustomizableEnum`, `DeletionConfiguration`, `FeatureConfiguration` would not work, because these object are not loaded by id reference.


#### Before-After results

1. Without and with L2 cache saving a case (infrastructure entities for germany). Result: Reduction of time from ~5.5s to ~0.6s.
<details>
<summary>CaseFacade.save: Performance comparison</summary>

![Screenshot 2022-11-15 at 15 21 03](https://user-images.githubusercontent.com/8166805/201931297-b4481426-76ba-471b-bd72-a49e3864a511.png)
![Screenshot 2022-11-15 at 15 21 26](https://user-images.githubusercontent.com/8166805/201931323-eb8ac6b8-a279-4a75-a0f6-1ac67387b9c8.png)

</details>

2. Load communities. Result: Slightly slower with L2 cache than before (~100ms difference)
<details>
<summary>CountryFacade.getAllActiveAsReference: Performance comparison</summary>

Without and with 2nd level cache getting communities of Bayern (infrastructure entities for germany)
![Screenshot 2022-11-15 at 15 25 57](https://user-images.githubusercontent.com/8166805/201931400-b589c137-800f-4d0b-ba25-aa07702005cf.png)
![Screenshot 2022-11-15 at 15 26 20](https://user-images.githubusercontent.com/8166805/201931392-e88c645c-b654-4e3e-8a6b-1b2b3f841190.png)

</details>

3. Generated sql when fetching one case with all ManyToOne relations but responsible region eager vs it being also lazy. Result: No joins on infrastructure.
<details>
<summary>SELECT case: SQL comparison</summary>

```sql
-- BEFORE
select case0_.id                                     as id1_10_0_,
       case0_.changeDate                             as changeda2_10_0_,
       case0_.change_user_id                         as change117_10_0_,
       case0_.creationDate                           as creation3_10_0_,
       case0_.uuid                                   as uuid4_10_0_,
       case0_.deleted                                as deleted5_10_0_,
       case0_.deletionReason                         as deletion6_10_0_,
       case0_.otherDeletionReason                    as otherdel7_10_0_,
       case0_.archiveUndoneReason                    as archiveu8_10_0_,
       case0_.archived                               as archived9_10_0_,
       case0_.endOfProcessingDate                    as endofpr10_10_0_,
       case0_.additionalDetails                      as additio11_10_0_,
       case0_.bloodOrganOrTissueDonated              as bloodor12_10_0_,
       case0_.caseAge                                as caseage13_10_0_,
       case0_.caseClassification                     as casecla14_10_0_,
       case0_.caseIdIsm                              as caseidi15_10_0_,
       case0_.caseIdentificationSource               as caseide16_10_0_,
       case0_.caseOfficer_id                         as caseof118_10_0_,
       case0_.caseOrigin                             as caseori17_10_0_,
       case0_.caseReferenceDefinition                as caseref18_10_0_,
       case0_.classificationComment                  as classif19_10_0_,
       case0_.classificationDate                     as classif20_10_0_,
       case0_.classificationUser_id                  as classi119_10_0_,
       case0_.clinicalConfirmation                   as clinica21_10_0_,
       case0_.clinicalCourse_id                      as clinic120_10_0_,
       case0_.clinicianEmail                         as clinici22_10_0_,
       case0_.cliniciandetails                       as clinici23_10_0_,
       case0_.clinicianPhone                         as clinici24_10_0_,
       case0_.community_id                           as commun121_10_0_,
       case0_.completeness                           as complet25_10_0_,
       case0_.contactTracingFirstContactDate         as contact26_10_0_,
       case0_.contactTracingFirstContactType         as contact27_10_0_,
       case0_.creationVersion                        as creatio28_10_0_,
       case0_.dengueFeverType                        as denguef29_10_0_,
       case0_.description                            as descrip30_10_0_,
       case0_.disease                                as disease31_10_0_,
       case0_.diseaseDetails                         as disease32_10_0_,
       case0_.diseaseVariant                         as disease33_10_0_,
       case0_.diseaseVariantDetails                  as disease34_10_0_,
       case0_.district_id                            as distri122_10_0_,
       case0_.districtLevelDate                      as distric35_10_0_,
       case0_.dontShareWithReportingTool             as dontsha36_10_0_,
       case0_.duplicateOf_id                         as duplic123_10_0_,
       case0_.endOfIsolationReason                   as endofis37_10_0_,
       case0_.endOfIsolationReasonDetails            as endofis38_10_0_,
       case0_.epiData_id                             as epidat124_10_0_,
       case0_.epidNumber                             as epidnum39_10_0_,
       case0_.epidemiologicalConfirmation            as epidemi40_10_0_,
       case0_.externalData                           as externa41_10_0_,
       case0_.externalID                             as externa42_10_0_,
       case0_.externalToken                          as externa43_10_0_,
       case0_.facilityType                           as facilit44_10_0_,
       case0_.followUpComment                        as followu45_10_0_,
       case0_.followUpStatus                         as followu46_10_0_,
       case0_.followUpStatusChangeDate               as followu47_10_0_,
       case0_.followUpStatusChangeUser_id            as follow125_10_0_,
       case0_.followUpUntil                          as followu48_10_0_,
       case0_.healthConditions_id                    as health126_10_0_,
       case0_.healthFacility_id                      as health127_10_0_,
       case0_.healthFacilityDetails                  as healthf49_10_0_,
       case0_.hospitalization_id                     as hospit128_10_0_,
       case0_.infectionSetting                       as infecti50_10_0_,
       case0_.internalToken                          as interna51_10_0_,
       case0_.investigatedDate                       as investi52_10_0_,
       case0_.investigationStatus                    as investi53_10_0_,
       case0_.laboratoryDiagnosticConfirmation       as laborat54_10_0_,
       case0_.maternalHistory_id                     as matern129_10_0_,
       case0_.nationalLevelDate                      as nationa55_10_0_,
       case0_.nosocomialOutbreak                     as nosocom56_10_0_,
       case0_.notACaseReasonDetails                  as notacas57_10_0_,
       case0_.notACaseReasonDifferentPathogen        as notacas58_10_0_,
       case0_.notACaseReasonNegativeTest             as notacas59_10_0_,
       case0_.notACaseReasonOther                    as notacas60_10_0_,
       case0_.notACaseReasonPhysicianInformation     as notacas61_10_0_,
       case0_.notifyingClinic                        as notifyi62_10_0_,
       case0_.notifyingClinicDetails                 as notifyi63_10_0_,
       case0_.outcome                                as outcome64_10_0_,
       case0_.outcomeDate                            as outcome65_10_0_,
       case0_.overwriteFollowUpUntil                 as overwri66_10_0_,
       case0_.person_id                              as person_67_10_0_,
       case0_.plagueType                             as plaguet68_10_0_,
       case0_.pointOfEntry_id                        as pointo130_10_0_,
       case0_.pointOfEntryDetails                    as pointof69_10_0_,
       case0_.portHealthInfo_id                      as porthe131_10_0_,
       case0_.postpartum                             as postpar70_10_0_,
       case0_.pregnant                               as pregnan71_10_0_,
       case0_.previousInfectionDate                  as previou72_10_0_,
       case0_.previousQuarantineTo                   as previou73_10_0_,
       case0_.prohibitionToWork                      as prohibi74_10_0_,
       case0_.prohibitionToWorkFrom                  as prohibi75_10_0_,
       case0_.prohibitionToWorkUntil                 as prohibi76_10_0_,
       case0_.quarantine                             as quarant77_10_0_,
       case0_.quarantineChangeComment                as quarant78_10_0_,
       case0_.quarantineExtended                     as quarant79_10_0_,
       case0_.quarantineFrom                         as quarant80_10_0_,
       case0_.quarantineHelpNeeded                   as quarant81_10_0_,
       case0_.quarantineHomePossible                 as quarant82_10_0_,
       case0_.quarantineHomePossibleComment          as quarant83_10_0_,
       case0_.quarantineHomeSupplyEnsured            as quarant84_10_0_,
       case0_.quarantineHomeSupplyEnsuredComment     as quarant85_10_0_,
       case0_.quarantineOfficialOrderSent            as quarant86_10_0_,
       case0_.quarantineOfficialOrderSentDate        as quarant87_10_0_,
       case0_.quarantineOrderedOfficialDocument      as quarant88_10_0_,
       case0_.quarantineOrderedOfficialDocumentDate  as quarant89_10_0_,
       case0_.quarantineOrderedVerbally              as quarant90_10_0_,
       case0_.quarantineOrderedVerballyDate          as quarant91_10_0_,
       case0_.quarantineReasonBeforeIsolation        as quarant92_10_0_,
       case0_.quarantineReasonBeforeIsolationDetails as quarant93_10_0_,
       case0_.quarantineReduced                      as quarant94_10_0_,
       case0_.quarantineTo                           as quarant95_10_0_,
       case0_.quarantineTypeDetails                  as quarant96_10_0_,
       case0_.rabiesType                             as rabiest97_10_0_,
       case0_.reInfection                            as reinfec98_10_0_,
       case0_.region_id                              as region132_10_0_,
       case0_.regionLevelDate                        as regionl99_10_0_,
       case0_.reinfectionDetails                     as reinfe100_10_0_,
       case0_.reinfectionStatus                      as reinfe101_10_0_,
       case0_.reportDate                             as report102_10_0_,
       case0_.reportLat                              as report103_10_0_,
       case0_.reportLatLonAccuracy                   as report104_10_0_,
       case0_.reportLon                              as report105_10_0_,
       case0_.reportingUser_id                       as report133_10_0_,
       case0_.responsibleCommunity_id                as respon134_10_0_,
       case0_.responsibleDistrict_id                 as respon135_10_0_,
       case0_.responsibleRegion_id                   as respon136_10_0_,
       case0_.screeningType                          as screen106_10_0_,
       case0_.sequelae                               as sequel107_10_0_,
       case0_.sequelaeDetails                        as sequel108_10_0_,
       case0_.sharedToCountry                        as shared109_10_0_,
       case0_.smallpoxLastVaccinationDate            as smallp110_10_0_,
       case0_.smallpoxVaccinationReceived            as smallp111_10_0_,
       case0_.smallpoxVaccinationScar                as smallp112_10_0_,
       case0_.sormasToSormasOriginInfo_id            as sormas137_10_0_,
       case0_.surveillanceOfficer_id                 as survei138_10_0_,
       case0_.symptoms_id                            as sympto139_10_0_,
       case0_.systemCaseClassification               as system113_10_0_,
       case0_.therapy_id                             as therap140_10_0_,
       case0_.trimester                              as trimes114_10_0_,
       case0_.vaccinationStatus                      as vaccin115_10_0_,
       case0_.wasInQuarantineBeforeIsolation         as wasinq116_10_0_,
       region1_.id                                   as id1_50_1_,
       region1_.changeDate                           as changeda2_50_1_,
       region1_.change_user_id                       as change_11_50_1_,
       region1_.creationDate                         as creation3_50_1_,
       region1_.uuid                                 as uuid4_50_1_,
       region1_.archived                             as archived5_50_1_,
       region1_.centrally_managed                    as centrall6_50_1_,
       region1_.area_id                              as area_id12_50_1_,
       region1_.country_id                           as country13_50_1_,
       region1_.epidCode                             as epidcode7_50_1_,
       region1_.externalID                           as external8_50_1_,
       region1_.growthRate                           as growthra9_50_1_,
       region1_.name                                 as name10_50_1_,
       area2_.id                                     as id1_4_2_,
       area2_.changeDate                             as changeda2_4_2_,
       area2_.change_user_id                         as change_u9_4_2_,
       area2_.creationDate                           as creation3_4_2_,
       area2_.uuid                                   as uuid4_4_2_,
       area2_.archived                               as archived5_4_2_,
       area2_.centrally_managed                      as centrall6_4_2_,
       area2_.externalId                             as external7_4_2_,
       area2_.name                                   as name8_4_2_,
       country3_.id                                  as id1_17_3_,
       country3_.changeDate                          as changeda2_17_3_,
       country3_.change_user_id                      as change_11_17_3_,
       country3_.creationDate                        as creation3_17_3_,
       country3_.uuid                                as uuid4_17_3_,
       country3_.archived                            as archived5_17_3_,
       country3_.centrally_managed                   as centrall6_17_3_,
       country3_.defaultName                         as defaultn7_17_3_,
       country3_.externalId                          as external8_17_3_,
       country3_.isoCode                             as isocode9_17_3_,
       country3_.subcontinent_id                     as subcont12_17_3_,
       country3_.unoCode                             as unocode10_17_3_,
       subcontine4_.id                               as id1_58_4_,
       subcontine4_.changeDate                       as changeda2_58_4_,
       subcontine4_.change_user_id                   as change_u9_58_4_,
       subcontine4_.creationDate                     as creation3_58_4_,
       subcontine4_.uuid                             as uuid4_58_4_,
       subcontine4_.archived                         as archived5_58_4_,
       subcontine4_.centrally_managed                as centrall6_58_4_,
       subcontine4_.continent_id                     as contine10_58_4_,
       subcontine4_.defaultName                      as defaultn7_58_4_,
       subcontine4_.externalId                       as external8_58_4_,
       continent5_.id                                as id1_16_5_,
       continent5_.changeDate                        as changeda2_16_5_,
       continent5_.change_user_id                    as change_u9_16_5_,
       continent5_.creationDate                      as creation3_16_5_,
       continent5_.uuid                              as uuid4_16_5_,
       continent5_.archived                          as archived5_16_5_,
       continent5_.centrally_managed                 as centrall6_16_5_,
       continent5_.defaultName                       as defaultn7_16_5_,
       continent5_.externalId                        as external8_16_5_
from cases case0_
         left outer join Region region1_ on case0_.responsibleRegion_id = region1_.id
         left outer join areas area2_ on region1_.area_id = area2_.id
         left outer join Country country3_ on region1_.country_id = country3_.id
         left outer join Subcontinent subcontine4_ on country3_.subcontinent_id = subcontine4_.id
         left outer join Continent continent5_ on subcontine4_.continent_id = continent5_.id
where case0_.id=?
```

```sql
-- AFTER
select case0_.id                                     as id1_10_0_,
       case0_.changeDate                             as changeda2_10_0_,
       case0_.change_user_id                         as change117_10_0_,
       case0_.creationDate                           as creation3_10_0_,
       case0_.uuid                                   as uuid4_10_0_,
       case0_.deleted                                as deleted5_10_0_,
       case0_.deletionReason                         as deletion6_10_0_,
       case0_.otherDeletionReason                    as otherdel7_10_0_,
       case0_.archiveUndoneReason                    as archiveu8_10_0_,
       case0_.archived                               as archived9_10_0_,
       case0_.endOfProcessingDate                    as endofpr10_10_0_,
       case0_.additionalDetails                      as additio11_10_0_,
       case0_.bloodOrganOrTissueDonated              as bloodor12_10_0_,
       case0_.caseAge                                as caseage13_10_0_,
       case0_.caseClassification                     as casecla14_10_0_,
       case0_.caseIdIsm                              as caseidi15_10_0_,
       case0_.caseIdentificationSource               as caseide16_10_0_,
       case0_.caseOfficer_id                         as caseof118_10_0_,
       case0_.caseOrigin                             as caseori17_10_0_,
       case0_.caseReferenceDefinition                as caseref18_10_0_,
       case0_.classificationComment                  as classif19_10_0_,
       case0_.classificationDate                     as classif20_10_0_,
       case0_.classificationUser_id                  as classi119_10_0_,
       case0_.clinicalConfirmation                   as clinica21_10_0_,
       case0_.clinicalCourse_id                      as clinic120_10_0_,
       case0_.clinicianEmail                         as clinici22_10_0_,
       case0_.cliniciandetails                       as clinici23_10_0_,
       case0_.clinicianPhone                         as clinici24_10_0_,
       case0_.community_id                           as commun121_10_0_,
       case0_.completeness                           as complet25_10_0_,
       case0_.contactTracingFirstContactDate         as contact26_10_0_,
       case0_.contactTracingFirstContactType         as contact27_10_0_,
       case0_.creationVersion                        as creatio28_10_0_,
       case0_.dengueFeverType                        as denguef29_10_0_,
       case0_.description                            as descrip30_10_0_,
       case0_.disease                                as disease31_10_0_,
       case0_.diseaseDetails                         as disease32_10_0_,
       case0_.diseaseVariant                         as disease33_10_0_,
       case0_.diseaseVariantDetails                  as disease34_10_0_,
       case0_.district_id                            as distri122_10_0_,
       case0_.districtLevelDate                      as distric35_10_0_,
       case0_.dontShareWithReportingTool             as dontsha36_10_0_,
       case0_.duplicateOf_id                         as duplic123_10_0_,
       case0_.endOfIsolationReason                   as endofis37_10_0_,
       case0_.endOfIsolationReasonDetails            as endofis38_10_0_,
       case0_.epiData_id                             as epidat124_10_0_,
       case0_.epidNumber                             as epidnum39_10_0_,
       case0_.epidemiologicalConfirmation            as epidemi40_10_0_,
       case0_.externalData                           as externa41_10_0_,
       case0_.externalID                             as externa42_10_0_,
       case0_.externalToken                          as externa43_10_0_,
       case0_.facilityType                           as facilit44_10_0_,
       case0_.followUpComment                        as followu45_10_0_,
       case0_.followUpStatus                         as followu46_10_0_,
       case0_.followUpStatusChangeDate               as followu47_10_0_,
       case0_.followUpStatusChangeUser_id            as follow125_10_0_,
       case0_.followUpUntil                          as followu48_10_0_,
       case0_.healthConditions_id                    as health126_10_0_,
       case0_.healthFacility_id                      as health127_10_0_,
       case0_.healthFacilityDetails                  as healthf49_10_0_,
       case0_.hospitalization_id                     as hospit128_10_0_,
       case0_.infectionSetting                       as infecti50_10_0_,
       case0_.internalToken                          as interna51_10_0_,
       case0_.investigatedDate                       as investi52_10_0_,
       case0_.investigationStatus                    as investi53_10_0_,
       case0_.laboratoryDiagnosticConfirmation       as laborat54_10_0_,
       case0_.maternalHistory_id                     as matern129_10_0_,
       case0_.nationalLevelDate                      as nationa55_10_0_,
       case0_.nosocomialOutbreak                     as nosocom56_10_0_,
       case0_.notACaseReasonDetails                  as notacas57_10_0_,
       case0_.notACaseReasonDifferentPathogen        as notacas58_10_0_,
       case0_.notACaseReasonNegativeTest             as notacas59_10_0_,
       case0_.notACaseReasonOther                    as notacas60_10_0_,
       case0_.notACaseReasonPhysicianInformation     as notacas61_10_0_,
       case0_.notifyingClinic                        as notifyi62_10_0_,
       case0_.notifyingClinicDetails                 as notifyi63_10_0_,
       case0_.outcome                                as outcome64_10_0_,
       case0_.outcomeDate                            as outcome65_10_0_,
       case0_.overwriteFollowUpUntil                 as overwri66_10_0_,
       case0_.person_id                              as person_67_10_0_,
       case0_.plagueType                             as plaguet68_10_0_,
       case0_.pointOfEntry_id                        as pointo130_10_0_,
       case0_.pointOfEntryDetails                    as pointof69_10_0_,
       case0_.portHealthInfo_id                      as porthe131_10_0_,
       case0_.postpartum                             as postpar70_10_0_,
       case0_.pregnant                               as pregnan71_10_0_,
       case0_.previousInfectionDate                  as previou72_10_0_,
       case0_.previousQuarantineTo                   as previou73_10_0_,
       case0_.prohibitionToWork                      as prohibi74_10_0_,
       case0_.prohibitionToWorkFrom                  as prohibi75_10_0_,
       case0_.prohibitionToWorkUntil                 as prohibi76_10_0_,
       case0_.quarantine                             as quarant77_10_0_,
       case0_.quarantineChangeComment                as quarant78_10_0_,
       case0_.quarantineExtended                     as quarant79_10_0_,
       case0_.quarantineFrom                         as quarant80_10_0_,
       case0_.quarantineHelpNeeded                   as quarant81_10_0_,
       case0_.quarantineHomePossible                 as quarant82_10_0_,
       case0_.quarantineHomePossibleComment          as quarant83_10_0_,
       case0_.quarantineHomeSupplyEnsured            as quarant84_10_0_,
       case0_.quarantineHomeSupplyEnsuredComment     as quarant85_10_0_,
       case0_.quarantineOfficialOrderSent            as quarant86_10_0_,
       case0_.quarantineOfficialOrderSentDate        as quarant87_10_0_,
       case0_.quarantineOrderedOfficialDocument      as quarant88_10_0_,
       case0_.quarantineOrderedOfficialDocumentDate  as quarant89_10_0_,
       case0_.quarantineOrderedVerbally              as quarant90_10_0_,
       case0_.quarantineOrderedVerballyDate          as quarant91_10_0_,
       case0_.quarantineReasonBeforeIsolation        as quarant92_10_0_,
       case0_.quarantineReasonBeforeIsolationDetails as quarant93_10_0_,
       case0_.quarantineReduced                      as quarant94_10_0_,
       case0_.quarantineTo                           as quarant95_10_0_,
       case0_.quarantineTypeDetails                  as quarant96_10_0_,
       case0_.rabiesType                             as rabiest97_10_0_,
       case0_.reInfection                            as reinfec98_10_0_,
       case0_.region_id                              as region132_10_0_,
       case0_.regionLevelDate                        as regionl99_10_0_,
       case0_.reinfectionDetails                     as reinfe100_10_0_,
       case0_.reinfectionStatus                      as reinfe101_10_0_,
       case0_.reportDate                             as report102_10_0_,
       case0_.reportLat                              as report103_10_0_,
       case0_.reportLatLonAccuracy                   as report104_10_0_,
       case0_.reportLon                              as report105_10_0_,
       case0_.reportingUser_id                       as report133_10_0_,
       case0_.responsibleCommunity_id                as respon134_10_0_,
       case0_.responsibleDistrict_id                 as respon135_10_0_,
       case0_.responsibleRegion_id                   as respon136_10_0_,
       case0_.screeningType                          as screen106_10_0_,
       case0_.sequelae                               as sequel107_10_0_,
       case0_.sequelaeDetails                        as sequel108_10_0_,
       case0_.sharedToCountry                        as shared109_10_0_,
       case0_.smallpoxLastVaccinationDate            as smallp110_10_0_,
       case0_.smallpoxVaccinationReceived            as smallp111_10_0_,
       case0_.smallpoxVaccinationScar                as smallp112_10_0_,
       case0_.sormasToSormasOriginInfo_id            as sormas137_10_0_,
       case0_.surveillanceOfficer_id                 as survei138_10_0_,
       case0_.symptoms_id                            as sympto139_10_0_,
       case0_.systemCaseClassification               as system113_10_0_,
       case0_.therapy_id                             as therap140_10_0_,
       case0_.trimester                              as trimes114_10_0_,
       case0_.vaccinationStatus                      as vaccin115_10_0_,
       case0_.wasInQuarantineBeforeIsolation         as wasinq116_10_0_
from cases case0_
where case0_.id=?
```

</details>

Ran into an issue after adding the ehcache dependency (@StefanKock if you have any idea how to fix this please let me know). Note that the library is not copied with the deploy-serverLibs although the other extra hibernate libs are there.

```
Caused by: org.hibernate.boot.registry.selector.spi.StrategySelectionException: Unable to resolve name [org.hibernate.cache.ehcache.internal.SingletonEhcacheRegionFactory] as strategy [org.hibernate.cache.spi.RegionFactory]
	at org.hibernate.boot.registry.selector.internal.StrategySelectorImpl.selectStrategyImplementor(StrategySelectorImpl.java:156)
	at org.hibernate.boot.registry.selector.internal.StrategySelectorImpl.resolveStrategy(StrategySelectorImpl.java:239)
	at org.hibernate.boot.registry.selector.internal.StrategySelectorImpl.resolveStrategy(StrategySelectorImpl.java:192)
	at org.hibernate.cache.internal.RegionFactoryInitiator.resolveRegionFactory(RegionFactoryInitiator.java:93)
	at org.hibernate.cache.internal.RegionFactoryInitiator.initiateService(RegionFactoryInitiator.java:47)
	at org.hibernate.cache.internal.RegionFactoryInitiator.initiateService(RegionFactoryInitiator.java:32)
	at org.hibernate.boot.registry.internal.StandardServiceRegistryImpl.initiateService(StandardServiceRegistryImpl.java:101)
	at org.hibernate.service.internal.AbstractServiceRegistryImpl.createService(AbstractServiceRegistryImpl.java:263)
	... 65 more
Caused by: org.hibernate.boot.registry.classloading.spi.ClassLoadingException: Unable to load class [org.hibernate.cache.ehcache.internal.SingletonEhcacheRegionFactory]
	at org.hibernate.boot.registry.classloading.internal.ClassLoaderServiceImpl.classForName(ClassLoaderServiceImpl.java:133)
	at org.hibernate.boot.registry.selector.internal.StrategySelectorImpl.selectStrategyImplementor(StrategySelectorImpl.java:152)
	... 72 more
Caused by: java.lang.ClassNotFoundException: Could not load requested class : org.hibernate.cache.ehcache.internal.SingletonEhcacheRegionFactory
	at org.hibernate.boot.registry.classloading.internal.AggregatedClassLoader.findClass(AggregatedClassLoader.java:210)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:589)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:398)
	at org.hibernate.boot.registry.classloading.internal.ClassLoaderServiceImpl.classForName(ClassLoaderServiceImpl.java:130)
	... 73 more
]]
```

--- a/sormas-backend/src/test/resources/logback-test.xml
+++ b/sormas-backend/src/test/resources/logback-test.xml
@@ -36,7 +36,8 @@
     <!-- Don't log KeycloakService warnings and info as it doesn't provide much information for tests -->
     <logger name="de.symeda.sormas.backend.user.KeycloakService" level="ERROR" />
 
-    <logger name="net.sf.ehcache" level="ERROR" />
+    <logger name="net.sf.ehcache" level="WARN" />
+    <logger name="net.sf.ehcache.config.ConfigurationFactory" level="ERROR" />
 
     <root level="error">
         <appender-ref ref="STDOUT" />

--- a/sormas-backend/src/test/resources/logback-test.xml
+++ b/sormas-backend/src/test/resources/logback-test.xml
@@ -36,7 +36,7 @@
     <!-- Don't log KeycloakService warnings and info as it doesn't provide much information for tests -->
     <logger name="de.symeda.sormas.backend.user.KeycloakService" level="ERROR" />
 
-    <logger name="net.sf.ehcache" level="WARN" />
+    <logger name="net.sf.ehcache" level="ERROR" />
 
     <root level="error">
         <appender-ref ref="STDOUT" />

--- a/sormas-ui/src/test/resources/logback-test.xml
+++ b/sormas-ui/src/test/resources/logback-test.xml
@@ -36,7 +36,8 @@
     <!-- Don't log KeycloakService warnings and info as it doesn't provide much information for tests -->
     <logger name="de.symeda.sormas.backend.user.KeycloakService" level="ERROR" />
 
-    <logger name="net.sf.ehcache" level="ERROR" />
+    <logger name="net.sf.ehcache" level="WARN" />
+    <logger name="net.sf.ehcache.config.ConfigurationFactory" level="ERROR" />
 
     <root level="error">
         <appender-ref ref="STDOUT" />

--- a/sormas-ui/src/test/resources/logback-test.xml
+++ b/sormas-ui/src/test/resources/logback-test.xml
@@ -36,7 +36,7 @@
     <!-- Don't log KeycloakService warnings and info as it doesn't provide much information for tests -->
     <logger name="de.symeda.sormas.backend.user.KeycloakService" level="ERROR" />
 
-    <logger name="net.sf.ehcache" level="WARN" />
+    <logger name="net.sf.ehcache" level="ERROR" />
 
     <root level="error">
         <appender-ref ref="STDOUT" />

